src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -0,0 +1,64 @@
+import { useCallback, useEffect, useState } from 'react';
+import { FiltersEnum } from '../../utils/FiltersEnum';
+import { TodoFilterProps } from '../../types/ComponentsProps';
+
+export const TodoFilter: React.FC<TodoFilterProps> = ({
+  todoList,
+  filterTodoList,
+  updateClearDisabled,
+}) => {
+  const [selectedFilter, setSelecetedFilter] = useState<FiltersEnum>(
+    FiltersEnum.All,
+  );
+
+  useEffect(() => {
```

Comment:
Instead of doing it here, you could do it on the parent component.
In this way, you would not need to pass _todoList_, _filterTodoList_, and _updateClearDisabled_ as props.
You could also pass this state to te parent
The only prop you would need is _setSelectedFilter_ and _selectedFilter_.